### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by rillig**
## Description
Clean up the code according to the inspections provided by Android Studio.
No functional change.

### CODE REVIEW

**Comment by rillig** [comment]
The results from the Codacy analysis are questionable.

* In SettingsLocalizationFragment.java I changed a single character, yet Codacy says something about "-36 clones removed".
* In UtilsGUI.java I didn't change whether the method parameter is reassigned, yet Codacy counts that as "+1" bad thing.

**Comment by UniQP** [comment]
Ok to test

**Comment by UniQP** [comment]
Hm, looks like the bot doesn't accept my command. @bekuno Can you give it a try.

**Comment by eddiemuc** [comment]
Ok to test

**Comment by Lineflyer** [comment]
ok to test

**Comment by Lineflyer** [comment]
I will check CI permissions

**Comment by Lineflyer** [comment]
eddiemuc had an typo in the jobs admin list, UniQP was missing. 
Added both.

File: src/main/java/menion/android/whereyougo/gui/activity/MainActivity.java [hunk 1]
```diff
@@ -1,22 +1,27 @@
...
  * Copyright (C) 2012 Menion <whereyougo@asamm.cz>
  */
 
 package menion.android.whereyougo.gui.activity;
 
+import static menion.android.whereyougo.permission.PermissionHandler.askAgainFor;
+import static menion.android.whereyougo.permission.PermissionHandler.checkKoPermissions;
+import static menion.android.whereyougo.permission.PermissionHandler.checkPermissions;
+import static menion.android.whereyougo.permission.PermissionHandler.needAskForPermission;
+
```
Comment (UniQP) [inline src/main/java/menion/android/whereyougo/gui/activity/MainActivity.java:24]: I would prefer the old import order (having the static imports after the corresponding non-static imports).

File: src/main/java/menion/android/whereyougo/MainApplication.java [hunk 2]
```diff
@@ -231,7 +229,7 @@ public void onCreate() {
             Log.e(TAG, "onCreate() - PANIC! Wipe out preferences", e);
             PreferenceManager.getDefaultSharedPreferences(this).edit().clear().commit();
         }
-    /* LEGECY SUPPORT -- END */
+        /* LEGECY SUPPORT -- END */
```
Comment (UniQP) [inline src/main/java/menion/android/whereyougo/MainApplication.java:232]:
```suggestion
        /* LEGACY SUPPORT -- END */
```

**Review by UniQP — COMMENTED** [review]
Looks good in general.

The only bigger problem I see is the code formatting. I see two options to proceed here:
1) Remove the code formatting commit (94bf8dc6e9c6a6cc2794ec299cedec36a670ee66) from this PR and open a separate PR for the code formatting (so that we can adapt the formatting configuration there to get the desired result).
2) Reduce the code formatting commit in a way that it only fixes obvious mistakes (i.e. remove all changes to multi-line expressions and import reordering). Optional: Open a separate formatting PR as in 1).

Which option do you prefer (or do you see another option)?

**Comment by rillig** [comment]
> Looks good in general.
> 
> The only bigger problem I see is the code formatting. I see two options to proceed here:
> 
> 1. Remove the code formatting commit ([94bf8dc](https://github.com/cgeo/WhereYouGo/commit/94bf8dc6e9c6a6cc2794ec299cedec36a670ee66)) from this PR and open a separate PR for the code formatting (so that we can adapt the formatting configuration there to get the desired result).
> 2. Reduce the code formatting commit in a way that it only fixes obvious mistakes (i.e. remove all changes to multi-line expressions and import reordering). Optional: Open a separate formatting PR as in 1).
> 
> Which option do you prefer (or do you see another option)?

When I applied the formatting, I assumed that all further development will happen in Android Studio, so I thought that choosing its default formatting options would be the best choice, as it deviates as little as possible from the standards that newcomers to the project would expect.

Do you have particular reasons why you want to deviate from these admittedly ad-hoc but still established standards?

By the way, I'm not finished yet with the cleanup. This first round was only about mechanical changes, without looking at the actual code. While doing that I already noticed several long expressions that were broken into lines as if they were a novel, not source code. Since I didn't want these formatting changes to affect the actual bug fixes and UX improvements, I had to do them first, before starting the interesting work.

**Comment by UniQP** [comment]
> When I applied the formatting, I assumed that all further development will happen in Android Studio, so I thought that choosing its default formatting options would be the best choice, as it deviates as little as possible from the standards that newcomers to the project would expect.
> 
> Do you have particular reasons why you want to deviate from these admittedly ad-hoc but still established standards?

Two concrete reasons are that it 1) also deviates from the existing code style (i.e. the established code style for this project) and 2) (IMO) makes some code less readable than it currently is.

We haven't fully configured automatic code formatting for any of the cgeo projects yet, so basically the existing code should be the guideline regarding code formatting. If you want automatic code formatting, the proper way forward would be to first create a code formatting configuration that reflects the current code style. This usually comes with a lot of discussion. Hence, my suggestion to move the commit into a separate PR, where we can solely focus the discussion on the code formatting.

**Comment by rillig** [comment]
> If you want automatic code formatting, the proper way forward would be to first create a code formatting configuration that reflects the current code style. This usually comes with a lot of discussion.

See https://github.com/cgeo/WhereYouGo/pull/387. If you are serious about "a lot of discussion", I'm outta here. I'm willing to invest some time into improving WhereYouGo (and to get to the interesting bugs, I first need the code to look consistent at least) but I won't take part in endless discussions.

**Comment by Lineflyer** [comment]
> See https://github.com/cgeo/WhereYouGo/pull/387. If you are serious about "a lot of discussion", I'm outta here.

I understand that.
Don't we have something in cgeo which we could apply also here, to limit the amount of discussion.
Furthermore no one is currently really actively working on this project. So if someone offers help but needs proper code formatting to be able to further work on it, why not as long as it follows some kind of known standard...just my few cents.

**Comment by Lineflyer** [comment]
@UniQP 
We recently merged code formatting configuration to the cgeo project.
Would it be applicable here as well or are there many difference to what is proposed by @rillig ?

**Comment by rillig** [comment]
First, this repository should be formatted according to the agreed-upon style.
Then, the remaining changes can be applied. Either you do it yourself or I do it.
I basically ran "Code | Inspect Code" on the whole project and picked one inspection after another, to keep each commit as simple and focused as possible.

**Comment by UniQP** [comment]
@rillig The code formatting PR is merged, so you can rebase this one.

**Comment by rillig** [comment]
I suggest ignoring Codacy for now, it is too whiny about code I touched for completely different reasons.
As long as I don't make the code worse than before, Codacy shouldn't complain at all.

**Comment by rillig** [comment]
The project seems to be inactive, I'm no longer interested in improving it.

**Pull request closed without merging**